File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
    const MOD = 1234567;
    const dp = Array(n + 1).fill(0);

    dp[0] = 1; // 0을 만드는 경우 1가지 (아무것도 선택하지 않음)
    dp[1] = 1; // 1을 만드는 경우 1가지 (1)

    for (let i = 2; i <= n; i++) {
        dp[i] = (dp[i - 1] + dp[i - 2]) % MOD;
    }

    return dp[n];
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    if n == 0:
        return 0
    if n == 1:
        return 1

    # dp[i] : i 계단까지 오르는 방법의 수
    dp = [0] * (n + 1)
    dp[0] = 1 
    dp[1] = 1  

    for i in range(2, n + 1):
        dp[i] = dp[i - 1] + dp[i - 2]  # 1칸 전 + 2칸 전 경우 합

    return dp[n] %1234567

